### PR TITLE
perf(hy_worldplay): longer video bench + HY model card

### PR DESCRIPTION
**Pull request by wenqingw-nv**
Follow-up to #155 for #203: GB300 perf re-bench, the `hy_worldplay` model card, and two `num_chunk≥4` fixes the larger GPU surfaced.

## Changes

- `_action.py`: unwrap the `torch.compile` `OptimizedModule` in `prefill_memory_kv_cache`; run memory-engaged chunks eager under CUDA graphs (the data-dependent memory `torch.cat` can't replay in a graph captured pre-memory). Both reachable only at `num_chunk≥4`.
- `bench.sh`: default `HY_VENDOR_SDPA=1` (vendor on cuDNN SDPA, matching native). `run.sh`: add `torchvision` to the vendor deps.
- `models/hy_worldplay.rst` (LingBot-style) + `perf-0530.md` + native sample videos; registered in the toctree; clean under `sphinx-build -W`.

## Bench

Native vs `wan/generate.py`: `num_chunk=8 / pose=w-31 / seed=0 / 704×1280`, warmup-discard 5, DiT+VAE scope, both legs cuDNN SDPA + `torch.compile`, single GB300. Post-warmup medians (chunks 5–7):

| stage | native | vendor | speedup |
| --- | ---: | ---: | ---: |
| DiT (diffuse) | 632 ms | 1206 ms | **1.91×** |
| VAE decode | 383 ms | 372 ms | ~parity |
| **DiT + VAE / chunk** | **1015 ms** | **1578 ms** | **1.55×** |

Per `data_local/` input:

| image | DiT nat/ven | VAE nat/ven | ratio | mean `\|Δ\|` |
| --- | ---: | ---: | ---: | ---: |
| `1.png` | 630 / 1208 | 382 / 374 | 1.56× | 40.8 |
| `2.png` | 632 / 1210 | 382 / 374 | 1.56× | 31.1 |
| `6.jpeg` | 628 / 1208 | 381 / 374 | 1.57× | 21.1 |
| `cat_surf.jpg` | 632 / 1207 | 381 / 375 | 1.56× | 51.8 |
| **median** | **631 / 1208** | **382 / 374** | **1.56×** | — |

Native ~1.91× on DiT, parity on VAE, input-independent (native DiT 628–632 ms across inputs). Parity `|Δ|` is cumulative bf16 AR drift (chunk-0 ≈14 matches #155's 12.91, ramps to ~49 by chunk-7, no jump at memory engagement); highest on the off-aspect `cat_surf` (625×350 upscaled).

## Native vs vendor video pairs

<table>
<tr><th>image</th><th>native</th><th>vendor</th></tr>
<tr><td><code>1.png</code></td><td><sub>

https://github.com/user-attachments/assets/eed5204d-b49d-42b4-96fa-d000a4252df6

</sub></td><td><sub>

https://github.com/user-attachments/assets/297fb31e-dfa6-4e73-a053-9504caf23760

</sub></td></tr>
<tr><td><code>2.png</code></td><td><sub>

https://github.com/user-attachments/assets/9e104ca6-1a95-487c-8936-cee653422dd9

</sub></td><td><sub>

https://github.com/user-attachments/assets/5dbb9d15-cffa-499f-9ab4-b42ee5613dda

</sub></td></tr>
<tr><td><code>6.jpeg</code></td><td><sub>

https://github.com/user-attachments/assets/92da7fcd-7776-4db0-82dc-0963d9ff0903

</sub></td><td><sub>

https://github.com/user-attachments/assets/99fac09c-eab7-4210-8b66-1fc373a9014d

</sub></td></tr>
<tr><td><code>cat_surf.jpg</code></td><td><sub>

https://github.com/user-attachments/assets/c2bbe1b6-f166-4476-abf0-8c822dc7eacb

</sub></td><td><sub>

https://github.com/user-attachments/assets/1e25cd40-ffdf-4c3a-be94-e112aceafb5a

</sub></td></tr>
</table>

### CODE REVIEW

**Comment by liruilong940607** [comment]
The visual results here in the native column is choppy -- this seems to indicate something is wrong with the current implementation. Needs revisit and fix.

Let's not add this to the public-facing doc until we are sure it is correctly implemented.

File: docs/source/_static/videos/hy_worldplay/hy-worldplay-hero.mp4
Comment (liruilong940607) [inline docs/source/_static/videos/hy_worldplay/hy-worldplay-hero.mp4:1]: Lets not check in videos into this repo. you can send me the videos on slack and I will host them.

File: integrations/hy_worldplay/hy_worldplay/_action.py [hunk 1]
```diff
@@ -953,18 +953,26 @@ def predict_flow(
...
             and len(input.memory_frame_indices) > 0
             and cache.clean_latent_history is not None
-            and cache.prefill_completed_for_chunk != ar_idx
-        ):
+        )
+        self._hy_memory_engaged = memory_engaged
+        # Run the reconstituted-context prefill once at the first
+        # denoising step of each chunk past the first; the
+        # ``prefill_completed_for_chunk`` latch suppresses re-runs on
+        # subsequent scheduler steps within the chunk.
```
Comment (liruilong940607) [inline integrations/hy_worldplay/hy_worldplay/_action.py:974]: This slipped through my previous review. Why would we need prefill_memory_kv? The `BlockKVCache` in flashdreams should naturally work with CUDAGraph.  

I traced it and it seems to lead to `HyWorldPlayMemoryKVCache`, which stored roped K and roped V. What's the reason we can't use `BlockKVCache` for that? And what's the reason it can't be compatible with cudagraph like other models?

Comment (wenqingw-nv) [reply integrations/hy_worldplay/hy_worldplay/_action.py:974]: `HyWorldPlayMemoryKVCache` is separate because memory isn't a rolling window, it's a per-chunk re-selection of non-contiguous frames, so BlockKVCache doesn't map directly. The non-graphability is just a stopgap; since the selected set is always exactly 16, a fixed-size in-place buffer is graph-compatible. This path also causes the choppy output, where the memory keys sit at collapsed RoPE positions [0..16) regardless of identity, so on the first window slide a frame's encoding shifts and the scene jolts. 

I'll rework it into a fixed-size, consistently-positioned in-place buffer to fix both, and numerically diff vs vendor's prefill first. Do you think we should: gather K/V from retained history or re-prefill each chunk (vendor re-prefills)?  @liruilong940607

File: docs/source/_static/videos/hy_worldplay/hy-worldplay-hero.mp4
Comment (wenqingw-nv) [reply docs/source/_static/videos/hy_worldplay/hy-worldplay-hero.mp4:1]: Done — removed the checked-in .mp4s from the repo; I'll send the clips over Slack for hosting.
